src/store/modules/product/actions.ts

```diff
@@ -93,6 +93,79 @@ const actions: ActionTree<ProductState, RootState> = {
     }
   },
 
+  // Update QueryString
+  async updateQueryString({ commit }, queryString) {
+    commit(types.PRODUCT_FILTERS_CURRENT_UPDATED, { 'filterName': 'queryString', 'value': queryString })
+  },
+
+  // Update Query
+  async updateQuery({ state, dispatch }, params) {
+    await dispatch('updateQueryString', params.queryString)
```

Comment:
I think this will reinitialise it to undefined for the updateProductFilters action

Reply:
Done Sir.